akamai/resource_akamai_cp_code.go

```diff
@@ -107,3 +112,53 @@ func resourceCPCodePAPINewCPCodes(d *schema.ResourceData, meta interface{}) *pap
 	}
 	return papi.NewCpCodes(contract, group)
 }
+
+// resourceCPCodeImport expects that the ID passed in will be CP Code ID, Contract ID and Group ID as a single tring delimited by colons: cpc_12345:ctr_12345:grp_12345
+func resourceCPCodeImport(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+	args, err := splitAndValidateCPCodeImportID(d.Id())
+	if err != nil {
+		return nil, err
+	}
+	cpCodeID := args[0]
+	contractID := args[1]
+	groupID := args[2]
+
+	cpCodes, err := papi.GetCpCodes(&papi.Contract{ContractID: contractID}, &papi.Group{GroupID: groupID})
+	if err != nil {
+		return nil, err
+	}
+	for _, c := range cpCodes.CpCodes.Items {
+		if c.CpcodeID == cpCodeID {
+			log.Printf("[DEBUG] Import CP Code: %+v", cpCodeID)
+			d.SetId(c.CpcodeID)
+			d.Set("name", c.CpcodeName)
+			d.Set("contract", contractID)
+			d.Set("group", groupID)
+			d.Set("product", c.ProductID)
+			return []*schema.ResourceData{d}, nil
+		}
+	}
+	return nil, fmt.Errorf("Akamai CP Code import failed. CP Code %s not found in Contract %s / Group %s", cpCodeID, contractID, groupID)
+}
+
+func splitAndValidateCPCodeImportID(id string) ([]string, error) {
+	args := strings.Split(id, ":")
+	if len(args) != 3 {
+		return nil, fmt.Errorf("Akamai CP Code import requires CP Code ID, Contract ID and Group ID delimited by ':'. Received %s", id)
+	}
```

Comment:
As convention, we use the comma character (`,`) as separator in the import ID.

```suggestion
	args := strings.Split(id, ",")
	if len(args) != 3 {
		return nil, fmt.Errorf("Akamai CP Code import requires CP Code ID, Contract ID and Group ID delimited by ','. Received %s", id)
	}
```